src/CSET/operators/transect.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2024 Met Office and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Operators to extract a transect given a tuple of xy coords to start/finish."""
+
+import logging
+
+import iris
+import numpy as np
+
+from CSET.operators._utils import get_cube_yxcoordname
+
+
+def calc_transect(cube: iris.cube.Cube, startxy: tuple, endxy: tuple):
+    """
+    Compute transect between startxy and endxy.
+
+    Computes a transect for a given cube containing at least latitude
+    and longitude coordinates, using an appropriate sampling interval along the
+    transect based on grid spacing. Also decides a suitable x coordinate to plot along
+    the transect - see notes for details on this.
+
+    Arguments
+    ---------
+
+    cube: Cube
+        An iris cube containing latitude and longitude coordinate dimensions,
+        to compute the transect on.
+    startxy: tuple
```

Comment:
TBH, I'm getting really confused with the x and y coordinates going through the code. Perhaps switching to lat and lon for coord names would be better?

Reply:
I've partially done this while refactoring the bounds checking.

Reply:
Have corrected all y_name -> lat_name and x_name -> lon_name in this script to make it easier to trace